drivers/lm75/lm75.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * A general driver for the lm75 temperature sensor including support for the
+ * lm75a and tmp1075 sensors as well.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "lm75.h"
+#include "lm75_regs.h"
+#include "lm75_params.h"
+#include "log.h"
+#include "kernel_defines.h"
+#include <stdint.h>
+#include <byteorder.h>
+#include "xtimer.h"
+
+#define I2C_BUS             dev->lm75_params.i2c_bus
+#define I2C_ADDR            dev->lm75_params.i2c_addr
+
+
+int lm75_init(lm75_t *dev, const lm75_params_t *params) {
+
+    /* Enable OS interrupt */
+    gpio_init_int(LM75_PARAM_INT, GPIO_IN_PU, GPIO_FALLING, __irq_os_cb, NULL);
```

Comment:
We only need to do that in `lm75_set_alarm()` right?
Also better get the GPIO parameter from `lm75_params_t`.

Reply:
The alarm function doesn't actually call any alarm, it only sets the temperature limits, i.e. Hysteresis/T-LOW and Over-temperature/T-HIGH temperatures. I will change the function name to better reflect this.

Reply:
Yes it configures the alarm threshold.
But we only need to configure the alarm GPIO if we set the alarm.